package config;

message CalendarConfig {
	optional int32 num_years = 1 [default = 30];
	repeated SpecialDay special_day = 2;

	optional double cell_size = 3;
	optional double cell_margin = 4;
	optional double font_size = 5;
	optional double bigger_font_size = 6;
	optional double line_width = 7;
	optional double month_label_height = 8;
	optional double year_label_width = 9;

	optional RGB rgb_header = 10;
	optional RGB rgb_header_sunday = 11;
	optional RGB rgb_month_line = 12;
	optional RGB rgb_holiday = 18;

	optional string bottom_left_label = 13;
	optional string bottom_right_label = 14;

	optional string header_font_family = 15;
	optional string number_font_family = 16;
	optional string quote_font_family = 17;
}

message SpecialDay {
	required int32 month = 1;
	required int32 day = 2;
	optional string svg = 3;
	optional int32 first_year = 4;
	optional int32 year = 5;
	optional RGB rgb = 6;
}

message RGB {
	required int32 red = 1;
	required int32 green = 2;
	required int32 blue = 3;
}
